meetings/2018-01-17/minutes.md

```diff
@@ -0,0 +1,417 @@
+tooling, and with that a strong willingness to contribute to efforts like this,
+so we should do everything possible to work together.
+
+### Scalac - Adriaan Moors, Lightbend
+
+Adriaan reiterates that tooling will be key to the next wave of Scala adoption,
+and wants to encourage the current efforts.
+
+Scalac originated as a batch compiler, and adapted into an incremental
+compiler. Dependencies for fine-grained compilation are different for Scala
+than for, say, Java. TASTY and SemanticDB are useful for exchanging that
+information. He's very happy to help enable the development of these features
+for Scalac.
+
+### Rsc - Eugene Burmako, Twitter
+
+[Rsc](http://goo.gl/LHmuym) is an experiment aiming to get 5-10x compilation
+speedups. It currently only supports a subset of Scala, and provides partial
+compilation, but is already showing impressive performance.
+
+Eugene believes that it will be possible to typecheck methods in parallel
+during compilation, and notes that separating name resolution and typechecking
+allows is essential for a good user experience. Rsc cross-compiles to Scala
+Native. Eugene spoke briefly about some future plans for Rsc, described in more
+detail in his slides.
+
+### Hydra - Iulian Dragos, Triplequotes
+
+Hydra is a Scala compiler that aims to be a drop-in replacement for Scalac,
+focussing on compile performance. Everything in Scalac from 2.11.8 - 2.12.4 is
+supported. The language-server front-end should not need specific knowledge of
+Hydra.
+
+Regarding building, special integration would be required, but it still aims to
+be a drop-in replcement for Scalac with a few additional options. Any build
+protocol would need to accommodate those additional parameters (e.g. specifying
+the number of cores to use for compilation).
+
+## Open discussion
+
+Jon noted that the open discussion was the least-structured part of the
+discussion, and should attempt to work out the scope of the project, rather
+than to solve the problems now, and advises everyone to avoid technical
+discussions.
+
+Jorge makes everyone aware that he and Olaf have been working on a build server
+protocol, the "BSP", which he sees as important for supporting certain
+features. The language server communicates with the build server and asks it to
+perform certain tasks, like compilation or requesting metadata. There is
+already a working integration with Bloop.
+
+Jon asks how the BSP could work with Pants and Bazel, and Jorge confirms that
+they should be compatible, as the BSP proposal has been expressly designed for
+this purpose.
+
+The BSP proposal has followed the "style" of the LSP protocol.
+
+Rory Graves made the point that different components have different
+responsibilities, and he encourages a modular approach with a "single point of
+entry" or a "busbar" for a number of different tools to communicate through,
+which holds the different components together.
+
```

Comment:
They key observation I was trying to make was that different components have and need difference subsets of the information.  The incremental compiler and SemanticDB both have forward information (X calls Y) when looking at X,  which works for things like 'go to definition).   
Completions require knowledge of the classpath and target type. (which for example the PC knows)
Find usages requires a reverse index (references in semantic db and the PC are forward), so you need to process the  raw information to do it.  
Incremental compilation requires a third subset of info - about file dependencies rather than class dependencies.  
So rather than having a thick component you want a thin shim which defers to information providers.  You also need cross linkage - e.g. when a compilation happens, notify the indexer to update the indexes.

Reply:
Definitely in agreement here. I think our only disagreement is which process receives the client connection... ie, which process is the hub.